pkg/plugins/manager/loader/initializer/initializer.go

```diff
@@ -64,7 +67,18 @@ func (i *Initializer) envVars(plugin *plugins.Plugin) []string {
 
 	hostEnv = append(hostEnv, i.awsEnvVars()...)
 	hostEnv = append(hostEnv, azsettings.WriteToEnvStr(i.cfg.Azure)...)
-	return getPluginSettings(plugin.ID, i.cfg).asEnvVar("GF_PLUGIN", hostEnv)
+
+	// Tracing
+	if i.cfg.Opentelemetry.IsEnabled() && !i.cfg.Opentelemetry.DisabledPlugins.IsDisabled(plugin.ID) {
+		hostEnv = append(
+			hostEnv,
+			fmt.Sprintf("GF_TRACING_OPENTELEMETRY_OTLP_ADDRESS=%s", i.cfg.Opentelemetry.Address),
+			fmt.Sprintf("GF_TRACING_OPENTELEMETRY_OTLP_PROPAGATION=%s", i.cfg.Opentelemetry.Propagation),
```

Comment:
If these are configured as environment variables when running Grafana they will be forwarded to plugins anyway since they match the existing configuration variables, e.g. https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#tracingopentelemetryotlp. We might need an environment variable telling if tracing is enabled/disabled for the plugin, e.g. `GF_PLUGIN_<plugin id>_TRACING=true/false`, and it will automatically support it to be configured thru Grafana config, e.g.
```
[plugin.<plugin id>]
tracing = true/false
```

Alternatively use other names for the variables. WDYT?

Reply:
Thanks for the heads up regarding the config via env var, I didn't think about it. However, if the user is using the config file rather than env vars, the plugin won't get the env var without those two lines, so I think we should either rename those passed to the plugin, or ensure that they are not repeated twice.

Regarding enabling/disabling under plugin settings:

I had that as a plugin setting in a previous iteration (now it's a list of plugin ids under `tracing.opentelemetry`), but I was worried it would clash with other user-defined plugin settings, so I moved it outside plugin settings.
If we make it a plugin setting we also don't really need the "DisabledPlugins" inside the tracing package (your first comment) and I agree it would make things clearer, and keep that logic in the plugin management.

Commit that changed this: https://github.com/grafana/grafana/pull/63714/commits/83357d09f1ef8368658322dc3963311903b71b77#diff-febafdf879d2ea02a4dd6ef6104b06facb6c5955cf4b6492890e5c2815b24cc7L72

I had the impression that the `[plugin.<PLUGIN ID>]` section was only supposed to contain user-defined values, that's why I decided to move it. If it's not a concern that it may clash with other user-defined stuff, I am more than happy to revert it back to how it was as I also would prefer having it under plugin settings :smile:

Reply:
> I had the impression that the `[plugin.<PLUGIN ID>]` section was only supposed to contain user-defined values, that's why I decided to move it. If it's not a concern that it may clash with other user-defined stuff, I am more than happy to revert it back to how it was as I also would prefer having it under plugin settings

No, not a concern. We already have a few things there I believe